packages/react-components/src/components/VideoGallery/utils/videoGalleryLayoutUtils.ts

```diff
@@ -41,21 +45,41 @@ const _useOrganizedParticipants = (props: OrganizedParticipantsArgs): OrganizedP
     maxRemoteVideoStreams,
     maxOverflowGalleryDominantSpeakers = DEFAULT_MAX_OVERFLOW_GALLERY_DOMINANT_SPEAKERS,
     isScreenShareActive = false,
-    pinnedParticipantUserIds = []
+    pinnedParticipantUserIds = [],
+    /* @conditional-compile-remove(gallery-layouts) */
+    layout
   } = props;
 
   const videoParticipants = remoteParticipants.filter((p) => p.videoStream?.isAvailable);
 
+  const participantsToSortTrampoline = (): VideoGalleryRemoteParticipant[] => {
+    /* @conditional-compile-remove(gallery-layouts) */
+    return layout !== 'speaker' ? videoParticipants : remoteParticipants;
+    return videoParticipants;
+  };
+
   visibleGridParticipants.current =
     pinnedParticipantUserIds.length > 0 || isScreenShareActive
       ? []
       : smartDominantSpeakerParticipants({
-          participants: videoParticipants,
+          participants: participantsToSortTrampoline(),
           dominantSpeakers,
           lastVisibleParticipants: visibleGridParticipants.current,
           maxDominantSpeakers: maxRemoteVideoStreams as number
         }).slice(0, maxRemoteVideoStreams);
 
+  /* @conditional-compile-remove(gallery-layouts) */
+  const dominantSpeakerToGrid =
```

Comment:
Can we add any unit tests for this logic?

Reply:
yup in PR following it in